gui/wxpython/startup/guiutils.py

```diff
@@ -309,15 +309,17 @@ def delete_mapset_interactively(guiparent, grassdb, location, mapset):
     if dlg.ShowModal() == wx.ID_YES:
         try:
             delete_mapset(grassdb, location, mapset)
+            dlg.Destroy()
+            return True
         except OSError as err:
             wx.MessageBox(
                 parent=guiparent,
                 caption=_("Error"),
                 message=_("Unable to delete mapset.\n\n%s") % err,
                 style=wx.OK | wx.ICON_ERROR | wx.CENTRE,
             )
-
-    dlg.Destroy()
+        dlg.Destroy()
+        return False
```

Comment:
This now gets executed only after Yes/OK, not after cancel. The implicit return returns `None` which  evaluates to `False` in the if-statement, so that's why it happens to work. The indentation is puts the lines into the if-statement, not the top level. If you place `print("XXX")` you can see that. Same for the other function.

Additionally, there is another return above in this function which returns `None`, so that's inconsistent. It needs to return `False`.